module/users/helper.js

```diff
@@ -610,32 +622,103 @@ module.exports = class UsersHelper {
    * @param {String} pageNo - Page number.
    * @param {String} pageSize - Page size.
    * @param {String} searchText - Search text.
+   * @param {String} userId - User Id.
    * @returns {Array} - Get user targeted programs.
    */
 
-  static programs(bodyData, pageNo, pageSize, searchText) {
+  static programs(bodyData, pageNo, pageSize, searchText, userId) {
     return new Promise(async (resolve, reject) => {
       try {
-        let targetedProgrms = await programsHelper.forUserRoleAndLocation(
+        // Here program details matching the user profile is retrieved.
+        let targetedPrograms = await programsHelper.forUserRoleAndLocation(
```

Comment:
@VISHNUDAS-tunerlabs If we are already getting whole data in line - 1014 of getUserJoinedProgramDetailsWithPreviousProfiles why do we need to query here, we can derive this from that